### PR TITLE
Application Security Group template

### PR DESCRIPTION
**Pull request by paulomarquesc**
### Changelog

* Initial commit of Application Security Groups, had to cancel previous PR since I realized I got the deploy to Azure icon links wrong.

### CODE REVIEW

File: 201-application-security-group/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        },
+        "adminPassword": {
+            "type": "securestring",
+            "metadata": {
+                "description": "Administrator pasword"
+            }
+        }
+    },
+    "variables": {
+        "nicApiVersion": "2017-09-01",
```
Comment (bmoore-msft) [inline 201-application-security-group/azuredeploy.json]: use literal apiVersions on the resources so schema validation can be done

File: 201-application-security-group/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        "vnetApiVersion": "2016-03-30",
+        "nsgApiVersion": "2017-09-01",
+        "vmExtensionApiVersion": "2015-06-15",
+        "asgApiVersion": "2017-10-01",
+        "vnetName": "vnet",
+        "vNetAddressSpace": "10.0.0.0/16",
+        "subnetName": "subnet01",
+        "subnetAdressPrefix": "10.0.0.0/24",
+        "vnetId": "[resourceId('Microsoft.Network/virtualNetworks',variables('vnetName'))]",
+        "subnetId": "[concat(variables('vnetId'),'/subnets/',variables('subnetName'))]",
```
Comment (bmoore-msft) [inline 201-application-security-group/azuredeploy.json]: remove concat and just use resourceId

File: 201-application-security-group/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        "pipName": "webServerPip",
+        "pipId": "[resourceId('Microsoft.Network/publicIPAddresses',variables('pipName'))]",
+        "imageInfo": {
+            "publisher": "OpenLogic",
+            "offer": "CentOS",
+            "sku": "6.9",
+            "version": "latest"
+        },
+        "vmStorageType": "Standard_LRS",
+        "scriptUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-application-security-group/install_nginx.sh"
```
Comment (bmoore-msft) [inline 201-application-security-group/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc) for how to handle artifacts

File: 201-application-security-group/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                        "[variables('vNetAddressSpace')]"
+                    ]
+                },
+                "subnets": [
+                    {
+                        "name": "[variables('subnetName')]",
+                        "properties": {
+                            "addressPrefix": "[variables('subnetAdressPrefix')]",
+                            "networkSecurityGroup": {
+                                "id": "[variables('nsgId')]"
```
Comment (bmoore-msft) [inline 201-application-security-group/azuredeploy.json]: does this need to be assigned on both the vnet and the asg?

File: 201-application-security-group/metadata.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+{
+    "itemDisplayName": "Application Security Groups",
```
Comment (bmoore-msft) [inline 201-application-security-group/metadata.json]: make this more descriptive of the scenario

**Comment by paulomarquesc** [comment]
Hi Brian, I implemented the required changes. Thanks.

**Pull request merged**